nemo/collections/asr/parts/submodules/jasper.py

```diff
@@ -462,72 +432,66 @@ def __init__(
         self.change_context_window(context_window=context_window)
 
         # Set default max sequence length
-        self.set_max_len(1)
+        self.max_len = 0
 
     def forward(self, x, lengths):
-        return self.forward_for_export(x, lengths)
-
-    def forward_for_export(self, x, lengths):
+        dtype = x.dtype
+        max_len = x.size(-1)
+        self.set_max_len(max_len)
         # The use of negative indices on the transpose allow for expanded SqueezeExcite
+        # Create sample mask - 1 represents value, 0 represents pad
+        mask = self.make_pad_mask(lengths, max_len)
+        mask = ~mask  # 0 represents value, 1 represents pad
         # Computes in float32 to avoid instabilities during training with AMP.
         with torch.cuda.amp.autocast(enabled=False):
-            # Create sample mask - 1 represents value, 0 represents pad
-            mask = self.make_pad_mask(lengths, max_audio_length=x.shape[-1], device=x.device)
-            mask = ~mask  # 0 represents value, 1 represents pad
-            x = x.float()  # For stable AMP, SE must be computed at fp32.
-            x.masked_fill_(mask, 0.0)  # mask padded values explicitly to 0
-            y = self._se_pool_step(x, mask)  # [B, C, 1]
-            y = y.transpose(1, -1)  # [B, 1, C]
-            y = self.fc(y)  # [B, 1, C]
-            y = y.transpose(1, -1)  # [B, C, 1]
+            # masked_fill needs to be here because of TRT issue with Where()
+            x = x.float().masked_fill(mask, 0.0)  # mask padded values explicitly to 0
+            y = self._se_pool_step(x, mask).to(x.dtype)
+
+        y = y.transpose(1, -1)  # [B, 1, C]
+        y = self.fc(y)  # [B, 1, C]            
+        y = y.transpose(1, -1)  # [B, C, 1]
 
         # Note: Keep for future, in case we improve WER from doing so.
         # if self.context_window >= 0:
         #     y = F.interpolate(y, size=x.shape[-1], mode=self.interpolation_mode)
 
         y = torch.sigmoid(y)
         y = x * y
-        return y, lengths
 
+        return y, lengths
+    
     def _se_pool_step(self, x, mask):
         # Negate mask back to represent 1 for signal and 0 for padded timestep.
         mask = ~mask
-
-        if self.context_window < 0:
+        if self.context_window < 0 or not self.training:
```

Comment:
Dont merge the two functionalities. I wanted two code paths for unlimited context window and limited context window, so that I can experiment with masked SE eval mode. Please revert this or not self.training. Mask has to be applied during both training and eval, and depends on context window.